e2e/framework.go

```diff
@@ -60,10 +60,73 @@ func initTracer(name string) *sdktrace.TracerProvider {
 }
 
 type cluster struct {
-	t      *testing.T
-	nodes  map[string]*node
-	tracer *sdktrace.TracerProvider
-	hook   transportHook
+	t               *testing.T
+	lock            sync.Mutex
+	nodes           map[string]*node
+	tracer          *sdktrace.TracerProvider
+	hook            transportHook
+	sealedProposals []*pbft.SealedProposal
+	proposer        pbft.NodeID
+}
+
+// getCurrentIndex returns a last index up to which there are inserted proposals
+func (c *cluster) getCurrentIndex() uint64 {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	idx := 0
+	if len(c.sealedProposals) > 0 {
```

Comment:
So if there are no items the index is '0' and if there is only one item the index is also 0?